File: app/src/main/java/com/smile/colorballs/Service/MyTop10ScoresService.java
package com.smile.colorballs.Service;

import android.app.Service;
import android.content.Intent;
import android.os.Bundle;
import android.os.IBinder;
import android.util.Log;
import android.util.Pair;

import androidx.annotation.Nullable;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import com.smile.colorballs.Constants;
import com.smile.smilelibraries.scoresqlite.ScoreSQLite;
import java.util.ArrayList;

public class MyTop10ScoresService extends Service {
    public final static String Action_Name = "com.smile.Service.MyTop10ScoresService";
    private final static String TAG = "MyTop10ScoresService";

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        Log.d(TAG, "onStartCommand() is called.");
        getDataAndSendBack();
        return super.onStartCommand(intent, flags, startId);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    private void getDataAndSendBack() {
        ArrayList<String> playerNames = new ArrayList<>();
        ArrayList<Integer> playerScores = new ArrayList<>();

        getLocalTop10Scores(playerNames, playerScores);
        Intent notificationIntent = new Intent(Action_Name);
        Bundle extras = new Bundle();
        extras.putStringArrayList(Constants.PlayerNamesKey, playerNames);
        extras.putIntegerArrayList(Constants.PlayerScoresKey, playerScores);
        notificationIntent.putExtras(extras);
        LocalBroadcastManager localBroadcastManager = LocalBroadcastManager.getInstance(getApplicationContext());
        localBroadcastManager.sendBroadcast(notificationIntent);
        Log.d(TAG, "getDataAndSendBack.sent result.");
        stopSelf();
        Log.d(TAG, "getDataAndSendBack.stopSelf().");
    }

    private boolean getLocalTop10Scores(ArrayList<String> playerNames, ArrayList<Integer> playerScores) {
        boolean status = true;
        playerNames.clear();
        playerScores.clear();
        try {
            ScoreSQLite scoreSQLiteDB = new ScoreSQLite(getApplicationContext());
            ArrayList<Pair<String, Integer>> resultList = scoreSQLiteDB.readTop10ScoreList();
            scoreSQLiteDB.close();
            for (Pair pair : resultList) {
                playerNames.add((String) pair.first);
                playerScores.add((Integer) pair.second);
            }
        } catch (Exception ex) {
            ex.printStackTrace();
            status = false;
        }
        Log.d(TAG, "getLocalTop10Scores.status = " + status);
        return status;
    }
}

File: app/src/main/java/com/smile/colorballs/service/MyTop10ScoresService.kt
package com.smile.colorballs.service

import android.app.Service
import android.content.Intent
import android.os.Bundle
import android.os.IBinder
import android.util.Log
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.smile.colorballs.Constants
import com.smile.smilelibraries.scoresqlite.ScoreSQLite

class MyTop10ScoresService : Service() {
    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        Log.d(TAG, "onStartCommand")
        getDataAndSendBack()
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    private fun getDataAndSendBack() {
        val playerNames = ArrayList<String>()
        val playerScores = ArrayList<Int>()
        getLocalTop10Scores(playerNames, playerScores)
        Intent(Action_Name).let {
            Bundle().apply {
                putStringArrayList(Constants.PlayerNamesKey, playerNames)
                putIntegerArrayList(Constants.PlayerScoresKey, playerScores)
                it.putExtras(this)
            }
            LocalBroadcastManager.getInstance(applicationContext).apply {
                Log.d(TAG, "getDataAndSendBack.sent result.")
                sendBroadcast(it)
            }
        }
        Log.d(TAG, "getDataAndSendBack.stopSelf().")
        stopSelf()
    }

    private fun getLocalTop10Scores(
        playerNames: ArrayList<String>,
        playerScores: ArrayList<Int>
    ): Boolean {
        var status = true
        playerNames.clear()
        playerScores.clear()
        try {
            ScoreSQLite(applicationContext).let {
                for (pair in it.readTop10ScoreList()) {
                    playerNames.add(pair.first as String)
                    playerScores.add(pair.second as Int)
                }
                it.close()
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
            status = false
        }
        Log.d(TAG, "getLocalTop10Scores.status = $status")
        return status
    }

    companion object {
        const val Action_Name = "com.smile.Service.MyTop10ScoresService"
        private const val TAG = "MyTop10ScoresService"
    }
}